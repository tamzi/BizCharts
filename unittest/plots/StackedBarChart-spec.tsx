import React from 'react';
import StackedBarChart from '../../src/plots/StackedBarChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  {
    label: 'Mon.',
    type: 'series1',
    value: 2800,
  },
  {
    label: 'Mon.',
    type: 'series2',
    value: 2260,
  },
  {
    label: 'Tues.',
    type: 'series1',
    value: 1800,
  },
  {
    label: 'Tues.',
    type: 'series2',
    value: 1300,
  },
  {
    label: 'Wed.',
    type: 'series1',
    value: 950,
  },
  {
    label: 'Wed.',
    type: 'series2',
    value: 900,
  },
  {
    label: 'Thur.',
    type: 'series1',
    value: 500,
  },
  {
    label: 'Thur.',
    type: 'series2',
    value: 390,
  },
  {
    label: 'Fri.',
    type: 'series1',
    value: 170,
  },
  {
    label: 'Fri.',
    type: 'series2',
    value: 100,
  },
];

describe('Plots-StackedBarChart', () => {
  test('堆叠条形图-旧版', () => {
    let chart = null;
    render(<StackedBarChart
      data={MOCK_DATA}
      xField="value"
      yField="label"
      stackField="type"
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
})
