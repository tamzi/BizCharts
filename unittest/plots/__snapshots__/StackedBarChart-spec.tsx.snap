// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-StackedBarChart 堆叠条形图-旧版 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "barWidthRatio": 0.6,
  "data": Array [
    Object {
      "label": "Mon.",
      "type": "series1",
      "value": 2800,
    },
    Object {
      "label": "Mon.",
      "type": "series2",
      "value": 2260,
    },
    Object {
      "label": "Tues.",
      "type": "series1",
      "value": 1800,
    },
    Object {
      "label": "Tues.",
      "type": "series2",
      "value": 1300,
    },
    Object {
      "label": "Wed.",
      "type": "series1",
      "value": 950,
    },
    Object {
      "label": "Wed.",
      "type": "series2",
      "value": 900,
    },
    Object {
      "label": "Thur.",
      "type": "series1",
      "value": 500,
    },
    Object {
      "label": "Thur.",
      "type": "series2",
      "value": 390,
    },
    Object {
      "label": "Fri.",
      "type": "series1",
      "value": 170,
    },
    Object {
      "label": "Fri.",
      "type": "series2",
      "value": 100,
    },
  ],
  "interactions": Array [
    Object {
      "type": "active-region",
    },
  ],
  "isStack": true,
  "legend": Object {
    "position": "top-left",
    "reversed": false,
  },
  "marginRatio": 0.03125,
  "renderer": "canvas",
  "seriesField": "type",
  "stackField": "type",
  "tooltip": Object {
    "offset": 20,
    "reversed": false,
    "shared": true,
    "showMarkers": false,
  },
  "xAxis": Object {
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "nice": true,
  },
  "xField": "value",
  "yAxis": Object {
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "label",
}
`;
